File: medium/topkfreq.py
class Solution:
    #let's create a dictionary {int:freq}, sort by the freq and return the first k
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        myDict ={}
        for num in nums:
            myDict[num] = myDict.get(num,1)+1

        myTuple = sorted(myDict.items(),key=lambda freq: freq[1],reverse=True)
        res = [x[0] for x in myTuple[:k]]
        return res

File: medium/topkfreq.cpp
class Solution
{
public:
    // let's create a map and sort it by the value, return first k
    vector<int> topKFrequent(vector<int> &nums, int k)
    {
        // int as key and freq as value
        map<int, int> mymap;

        for (auto it = nums.begin(); it != nums.end(); it++)
        {
            mymap[*it]++;
        }
        vector<pair<int, int>> pairs;
        for (auto it = mymap.begin(); it != mymap.end(); it++)
        {
            pairs.push_back(*it);
        }
        sort(pairs.begin(), pairs.end(), [](const auto &a, const auto &b)
             { return a.second > b.second; });
        vector<int> res;
        for (auto it = pairs.begin(); it != pairs.begin() + k; it++)
        {
            res.push_back(it->first);
        }
        return res;
    }
};